src/internet/model/tcp-bbr-v2.cc

```diff
@@ -679,6 +692,19 @@ TcpBbrV2::GetCwndGain()
     return m_cWndGain;
 }
 
+uint32_t
+TcpBbrV2::GetBDP(Ptr<TcpSocketState> tcb, double gain)
+{
+    NS_LOG_FUNCTION(this << tcb << gain);
+    if (m_minRtt == Time::Max())
+    {
+        return tcb->m_initialCWnd * tcb->m_segmentSize;
+    }
+
+    double estimatedBdp = m_maxBwFilter.GetBest() * m_minRtt / 8.0;
+    return estimatedBdp * gain;
+}
```

Comment:
`BDP = max BW * min RTT * gain`.
`m_maxBwFilter.GetBest()` returns value in bits, so `/ 8.0` is for converting to bytes.